### ISSUE TITLE
XSD incorrectly states the default value for the mongo-ref attribute of the mongo:db-factory configuration element [DATAMONGO-456]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-mongodb#1393]
**[Matt Senter](https://jira.spring.io/secure/ViewProfile.jspa?name=kungfuters)** opened **[DATAMONGO-456](https://jira.spring.io/browse/DATAMONGO-456?redirect=false)** and commented

There is a disconnect between the full Spring Mongo documentation and `spring-mongo-1.0.xsd` (also `spring-mongo-1.1.xsd`.) Essentially, the XSD says that when declaring a `<mongo:db-factory>` element, the default value for the `mongo-ref` attribute is `mongo`, implying that if you declare no `mongo-ref` attribute, the configuration will reference any existing `<mongo:mongo />` element for further configuration of mongo. However, that is not the case. As the full documentation states, you must declare this mongo-ref **if** you want to use another mongo instance (with advanced options for example) instead of the one configured by `<mongo:db-factory />`.

tl;dr - The XSD description of the `db-factory` element should be changed from "The reference to a Mongo. Will default to `mongo`." to something more accurate, such as "The reference to a Mongo. Set this value if you need to create a more advanced Mongo beyond the capabilities of the db-factory element." =)

---

**Affects:** 1.0.1, 1.1 M1

**Referenced from:** commits https://github.com/spring-projects/spring-data-mongodb/commit/3661b2981e7d5481876c12299fe3e0ff8d5de2d0, https://github.com/spring-projects/spring-data-mongodb/commit/4ca16f8d15203857c6f70353e43c707d1033e986
